src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvAwaitable.cs

```diff
@@ -49,10 +49,16 @@ public UvWriteResult GetResult()
 
         public void OnCompleted(Action continuation)
         {
+            // There should never be a race between IsCompleted and OnCompleted since both operations
+            // should always be on the libuv thread
```

Comment:
That plus we already assert before every operation that we're on the libuv thread. https://github.com/aspnet/KestrelHttpServer/blob/42d82a507da9c7522ae64c1a4c5cdd6556035162/src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/Networking/UvMemory.cs#L84